scripts/solutions.sh

```diff
@@ -0,0 +1,479 @@
+}
+
+namespace_is_in_environment() {
+    local -r namespace=$1 environment=$2
+
+    kubectl get namespace --selector="$ENV_LABEL=$environment" \
+        --output=jsonpath='{ .items[*].metadata.name }' \
+        | grep --quiet --word-regexp "$namespace"
+}
+
+check_namespace() {
+    if ! kubectl get namespace "$NAMESPACE" &> /dev/null; then
+        echo 1>&2 "Namespace '$NAMESPACE' does not exist"
+        return 1
+    fi
+
+    if ! namespace_is_in_environment "$NAMESPACE" "$NAME" &> /dev/null; then
+        echo 1>&2 "Namespace '$NAMESPACE' is not linked to the"
+            "Environment '$NAME'"
+        return 1
+    fi
+}
+
+create_environment() {
+    if ! [[ ${NAMESPACE:-} ]]; then
+        NAMESPACE=$NAME
+        if ! check_namespace_validity "$NAMESPACE"; then
+            echo 1>&2 "Environment name '$NAME' can't be used as a Namespace" \
+                "name. Please provide a Namespace name using the --namespace" \
+                "option, which only contains lower case alphanumeric" \
+                "characters or '-', and starts and ends with an alphanumeric" \
+                "character."
+                return 1
+        fi
+    fi
+
+    if kubectl get namespace "$NAMESPACE" &> /dev/null; then
+        echo 1>&2 "Namespace '$NAMESPACE' already exists"
+        return 1
+    fi
+
+    run "Creating Namespace '$NAMESPACE'" \
+        kubectl create namespace "$NAMESPACE"
+    run "Adding Namespace '$NAMESPACE' to Environment '$NAME'" \
+        kubectl label namespace "$NAMESPACE" "$ENV_LABEL=$NAME"
+    run "Setting Namespace '$NAMESPACE' description" \
+        kubectl annotate namespace "$NAMESPACE" \
+        "$ENV_ANNOTATION=${DESCRIPTION:-}"
+    run "Creating Namespace '$NAMESPACE' configuration" \
+        kubectl create configmap "$ENV_CONFIGMAP" \
+        --namespace "$NAMESPACE"
+}
+
+delete_namespaces() {
+    for namespace; do
+        run "Deleting Namespace '$namespace'" \
+            kubectl delete namespace "$namespace" &> /dev/null
+    done
+}
+
+delete_environment() {
+    local -a namespaces=()
+
+    if ! [[ ${NAMESPACE:-} ]]; then
+        namespaces=($( \
+            kubectl get namespace \
+            --selector="$ENV_LABEL=$NAME" \
+            --output=jsonpath='{ .items[*].metadata.name }'
+        ))
+        if ! (( ${#namespaces[@]} )); then
+            echo 1>&2 "No such Environment '$NAME'"
+            return 1
+        fi
+    else
+        check_namespace || return
+        namespaces=("$NAMESPACE")
+    fi
+
+    delete_namespaces "${namespaces[@]}"
+}
+
+check_solution_exists() {
+    local solution_data
+
+    solution_data=$(
+        kubectl get configmap "$SOLUTIONS_CONFIGMAP" \
+            --namespace "$SOLUTIONS_NAMESPACE" \
+            --output=jsonpath="{ .data['$SOLUTION'] }"
+    )
+
+    if ! [[ "$solution_data" ]]; then
+        echo 1>&2 "Solution '$SOLUTION' is not deployed"
+        return 1
+    fi
+
+    if ! jq -e ".[] | select(.version == \"$VERSION\")" \
```

Comment:
Installing a package just for this single line seems a bit heavy... But that's OK I guess, `jq` is a nice tool to have at hand, could even make sense to put it in `metalk8s-utils` container image (see #2156).

FWIW, here's how you could have done it without external deps:

```shellscript
if ! python -c "import json, sys;
      available = any(v["version"] == "$VERSION" for v in json.loads('$solution_data'));
      sys.exit(0 if available else 1)"; then
```

Reply:
Yeah, it's a bit heavy, but I think `jq` can still be useful to parse json as I did here, if one want to read the content of a configmap for example.

Reply:
I've removed it and used the python solution.